12.0 mig account journal lock date

blaggacao opened the pull request.

sbidoul committed:
[ADD] account_journal_lock_date

sbidoul committed:
[FIX] account_journal_lock_date test on travis

The test failed on travis (not on a local install) because the
admin users if part of the Adviser group. Make sure we have the right
group and add a test for that feature too.

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza commented:
Congratulations on your first module migration.

Check CIs status. Maybe dependencies are not yet migrated?

You can squash together adjacent "OCA Transbot.." commits using https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests

And finally, a new method for generating the README easily out of fragments is available with https://github.com/OCA/maintainer-tools/#readme-generator

Commit:
[UPD] README

blaggacao commented:
:+1: Is this an automated message ? :smile:

blaggacao commented:
>You can squash together adjacent "OCA Transbot.." commits using https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests

Kind of a case for an `oca-squash-translation-commits` helper? :wink:

pedrobaeza commented:
No, I do it manually. Some people suspects that I have minions working for me, but I can assure it's me, hahaha. It's just my "extreme GTD" technique that annoys you sometimes :wink:

blaggacao commented:
@pedrobaeza Every time less so :wink: I'd like to pre-validate something of importance (I feel) about OCA with you, is there a back channel where I can reach out to you?

pedrobaeza commented:
> Kind of a case for an oca-squash-translation-commits helper? wink

Yeah, it can be, but this is not going to happen anymore with Weblate. The idea is to remove from each version unneeded iteration commits. That can happen with Transifex, or incremental translations on Weblate, or even old Launchpad system, or several commit from same author that are now meaningless from being split, but of course always preserving authors. With this, on 13.0 we will have very clean branches that don't take a lot to be pulled and the rest of git operations.

Commit:
[REM] remove pdependency (functional preference)

pedrobaeza commented:
> I'd like to pre-validate something of importance (I feel) about OCA with you, is there a back channel where I can reach you?

Asynchronously you can contact by mail in my corporate mail, and I'm also in hangouts with that one for synchronous chats, but that's more difficult due to timeslots.

Commit:
[IMP] Improve readme

Commit:
[FIX] dependency

Commit:
[IMP] description

Commit:
[IMP] journal lock user semantics

The existing implementaiton already provided a "positive offset from period
lock date" semantic. However the UX design & description would have made the
user beleive to be able to specify any lock date on the journal that would
ignore the more general lock dates.

This is neither implemented nor intended. Therefore, improve the UX, so this
behaviour is clearly communicated and proper onchange, warnings, and errors are
generated for the user.

Defining an offset also makes the need for wizard obsolete alltogether, which
is why the Roadmap is updated accordingly.

Signed-off-by: "David Arnold" <"dar@xoe.solutions">

Commit:
[IMP] handle lock date updates

Validate for no draft moves on tracing the lock date on an individual journal.

Commit:
[FIX] lint & copyright

Commit:
[FIX] tests & improve them

Commit:
Lint and UX

nikul-serpentcs commented on account_journal_lock_date/__manifest__.py:
```diff
@@ -0,0 +1,19 @@
+    'version': '12.0.1.0.0',
+    'license': 'AGPL-3',
+    'author': 'ACSONE SA/NV,Odoo Community Association (OCA)',
+    'website': 'https://acsone.eu/',
```
@blaggacao Should be `https://github.com/OCA/account-financial-tools`

nikul-serpentcs commented on account_journal_lock_date/__manifest__.py:
```diff
@@ -0,0 +1,19 @@
+    'data': [
+        'views/account_journal.xml',
+    ],
+    'demo': [
```
@blaggacao If nor any file then remove it

nikul-serpentcs commented on account_journal_lock_date/views/account_journal.xml:
```diff
@@ -0,0 +1,39 @@
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+
+<odoo>
+<data>
```
@blaggacao remove `<data>`

nikul-serpentcs commented on account_journal_lock_date/models/account_journal.py:
```diff
@@ -0,0 +1,82 @@
+    _inherit = 'account.journal'
+
+    journal_lock_date_period_offset = fields.Integer(
+        string="Lock date offset", default=0,
```
@blaggacao no need to `default = 0`

https://odoo-new-api-guide-line.readthedocs.io/en/latest/fields.html#integer

nikul-serpentcs commented on account_journal_lock_date/models/account_journal.py:
```diff
@@ -0,0 +1,82 @@
+    )
+    has_period_lock_date = fields.Boolean(compute='_compute_lock_date')
+    journal_lock_date = fields.Date(
+        string="Lock date", readonly=True, store=True,
```
@blaggacao no need to `readonly=True`, Compute always ` readonly`

nikul-serpentcs reviewed (changes requested):
@blaggacao Improve Code

pedrobaeza commented:
As @blaggacao has moved to other software AFAIK, and there's #831 from active contributor, we are moving to use that one if you don't mind. If you don't agree, just tell me, as you have priority being the first PR. This module had a problem in consistency with the name, as it should be independent from permanent lock.
